Source/Core/Core/PowerPC/BreakPoints.h

```diff
@@ -132,3 +134,39 @@ class MemChecks
   Core::System& m_system;
   bool m_mem_breakpoints_set = false;
 };
+
+class DelayedMemCheckUpdate final
+{
+public:
+  DelayedMemCheckUpdate(MemChecks* memchecks, bool update_needed = false)
+      : m_memchecks(memchecks), m_update_needed(update_needed)
+  {
+  }
+
+  DelayedMemCheckUpdate(const DelayedMemCheckUpdate&) = delete;
+  DelayedMemCheckUpdate(DelayedMemCheckUpdate&& other) = delete;
+  DelayedMemCheckUpdate& operator=(const DelayedMemCheckUpdate&) = delete;
+  DelayedMemCheckUpdate& operator=(DelayedMemCheckUpdate&& other) = delete;
+
+  ~DelayedMemCheckUpdate()
+  {
+    if (m_update_needed)
+      m_memchecks->Update();
+  }
+
+  DelayedMemCheckUpdate& operator|=(DelayedMemCheckUpdate&& other)
+  {
+    if (m_memchecks == other.m_memchecks)
+    {
+      m_update_needed |= other.m_update_needed;
+      other.m_update_needed = false;
+    }
+    return *this;
+  }
+
+  operator bool() const { return m_update_needed; }
```

Comment:
Is this `operator bool` needed? Other than that, LGTM.

Reply:
I can't find anything that's actually using it, but I figured it would make sense to preserve the functionality of being able to tell whether the call to Remove did in fact remove something.